### PR TITLE
feat: add tsdoc documentation to Config

### PR DESCRIPTION
**Pull request by harbu**
## Summary

Add tsdoc documentation to main `Config` object.

## Checklist before requesting a review

- [x] Is this a breaking change? If it is, be clear in summary.
- [x] Read through code myself one more time.
- [x] Make sure any and all `TODO` comments left behind are meant to be left in.
- [x] Has reasonable passing test coverage?
- [x] Updated changelog if applicable.
- [x] Updated documentation if applicable.

### CODE REVIEW

File: packages/client/src/Config.ts [hunk 1]
```diff
@@ -50,64 +50,241 @@ export interface EthereumNetworkConfig {
...
+         *
+         * In streams with 1000+ publishers, it is important to limit the amount
+         * of control message traffic that gets generated to avoid network buffers
+         * from overflowing.
+         */
         maxKeyRequestsPerSecond?: number
     }
 
+    /**
+     * These settings apply to the network node of the client.
```
Comment (fonty1) [inline packages/client/src/Config.ts]:
```suggestion
     * These settings determine how the client performs and interacts with the Streamr Network.
```

Comment (fonty1) [reply packages/client/src/Config.ts]: Not exactly sure how to best describe this section, I was thrown off a bit with "network node of the client" though

File: packages/client/src/Config.ts [hunk 2]
```diff
@@ -50,64 +50,241 @@ export interface EthereumNetworkConfig {
...
+        /**
+         * The maximum outgoing message size accepted by WebRTC connections.
+         * Messages exceeding the maximum size are simply discarded.
+         */
         webrtcMaxMessageSize?: number
+
+        /**
+         * Defines a custom UDP port range to be used for WebRTC connections.
+         * This port range should not be restricted by enclosing firewalls
+         * or VPC configurations.
```
Comment (fonty1) [inline packages/client/src/Config.ts]: Virtual Port-Channel or Virtual private cloud? Probs best to add the full name here for searchability also.

File: packages/client/src/Config.ts [hunk 3]
```diff
@@ -50,64 +50,241 @@ export interface EthereumNetworkConfig {
...
 
-    /** Attempt to order messages */
+    /**
+     * Set to true to enable message ordering.
+     *
+     * Due to the distributed nature of the network, messages may occasionally
+     * arrive to the client out-of-order. By enabling message ordering, the
+     * client will reorder received messages to the intended order for you.
+     *
+     * */
```
Comment (teogeb) [inline packages/client/src/Config.ts]: The first line is kind of trivial (guessable from the field name). We could consider if it is more straightforward to write this without introducing thee extra term. 

I.e. skip the middle step in this chain: "I want to client to reorder messages" -> "I need to enable message ordering" -> "I set this to `true`".

```suggestion
    /**
  
     * Due to the distributed nature of the network, messages may occasionally
     * arrive to the client out-of-order. Set this option to `true` if you want that the 
     * client reorders the received messages to the intended order.
     *
     * */
```

File: packages/client/src/Config.ts [hunk 4]
```diff
@@ -50,64 +50,241 @@ export interface EthereumNetworkConfig {
...
-    retryResendAfter?: number
+
+    /**
+     * When gap filling is enabled and a gap is encountered, this option
+     * defines the amount of time to wait before attempting to _actively_ fill
+     * in the gap.
+     *
+     * Rationale: data may just be arriving out-of-order and the missing
+     * message(s) may be on their way. For efficiency, it makes sense to wait a
+     * little before actively attempting to fill in a gap, as this involves
```
Comment (teogeb) [inline packages/client/src/Config.ts:97]: Could move this description of active filling to `maxGapRequest` section as the concept of active filling is coined there for the first time.

File: packages/client/src/Config.ts [hunk 5]
```diff
@@ -50,64 +50,241 @@ export interface EthereumNetworkConfig {
...
+     * a resend request / response interaction with a storage node.
+     */
     gapFillTimeout?: number
 
     /**
-     * Message encryption/decryption
+     * When gap filling is enabled and a gap is encountered, a resend request
```
Comment (teogeb) [inline packages/client/src/Config.ts:103]: This is not always done, only after `gapFillTimeout`. 

We could say something like "If we make a storage node request in order to fill a gap, and the request fails, the client can retry the fetch from (another?) storage. This option controls how long to wait for a storage node response from the storage node before proceeding to the next attempt.

The number is in milliseconds."

(The unit of time, i.e. milliseconds, could be included in each of the sections where applicable: `retryResendAfter`, `keyRequestTimeout`, `newWebrtcConnectionTimeout`).

File: packages/client/src/Config.ts [hunk 6]
```diff
@@ -50,64 +50,241 @@ export interface EthereumNetworkConfig {
...
+    /**
+     * Controls how messages encryption and decryption should be handled and
+     * how encryption keys should be exchanged.
      */
     encryption?: {
         /**
          * Enable experimental Lit Protocol key exchange.
          *
-         * When enabled encryption key storing and fetching will be primarily done through the Lit Protocol and
+         * When enabled encryption key storing and fetching will primarily be done through the Lit Protocol and
```
Comment (teogeb) [inline packages/client/src/Config.ts]: Link to LIT protocol web site?

File: packages/client/src/Config.ts [hunk 7]
```diff
@@ -50,64 +50,241 @@ export interface EthereumNetworkConfig {
...
     network?: {
+        /**
+         * The network-wide identifier of this node. Should be unique
+         * within the Streamr Network.
+         */
         id?: string
+
+        /**
+         * Whether to accept proxy connections. Enabling this option allows
+         * the network node to act as proxy on behalf of other nodes / clients.
```
Comment (teogeb) [inline packages/client/src/Config.ts]:
```suggestion
         * this network node to act as proxy on behalf of other nodes / clients.
```

File: packages/client/src/Config.ts [hunk 8]
```diff
@@ -50,64 +50,241 @@ export interface EthereumNetworkConfig {
...
         disconnectionWaitTime?: number
+
+        /**
+         * Defines how often to ping connected nodes to determine connection
+         * aliveness.
+         */
         peerPingInterval?: number
+
+        /**
+         * Determines how often, at most, to include RTT statistics in status
```
Comment (teogeb) [inline packages/client/src/Config.ts]:
```suggestion
         * Determines how often, at most, to include round-trip time statistics in status
```

File: packages/client/src/Config.ts (hunk 1 above)
Comment (harbu) [reply packages/client/src/Config.ts]: Well... there is a network node within the client and these settings are passed to it when constructing it. However, maybe the user doesn't want to know this level of detail so I think your suggestion makes sense here.

File: packages/client/src/Config.ts (hunk 5 above)
Comment (harbu) [reply packages/client/src/Config.ts:103]: Changed wording to emphasize conditionality of resend request:
> When gap filling is enabled and a gap is encountered, a resend request
_may eventually_ be sent ...

Comment (harbu) [reply packages/client/src/Config.ts:103]: I would leave out the details "of requesting from another storage node".

File: packages/client/src/Config.ts (hunk 4 above)
Comment (harbu) [reply packages/client/src/Config.ts:97]: There is no guarantee of a reading order here so each section should sort of stand on its own (to the extent that this is possible without excessive repeating). If anything I would imagine the `gapFill` option to be referenced first.

File: packages/client/src/Config.ts (hunk 5 above)
Comment (harbu) [reply packages/client/src/Config.ts:103]: Added units to descriptions

**Pull request merged**